### PR TITLE
thrift_proxy: remove well_known_names

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long Dai <long0dai@foxmail.com>

Commit Message: 

Remove well_known_names in thrift_proxy field, part of #7238

Additional Description:
Risk Level: Low
Testing: functional
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc [hunk 1]
```diff
@@ -338,7 +338,7 @@ TEST_F(ThriftRateLimitFilterTest, ErrorResponseWithDynamicMetadata) {
   EXPECT_CALL(filter_callbacks_.stream_info_, setDynamicMetadata(_, _))
       .WillOnce(Invoke([&dynamic_metadata](const std::string& ns,
                                            const ProtobufWkt::Struct& returned_dynamic_metadata) {
-        EXPECT_EQ(ns, ThriftProxy::ThriftFilters::ThriftFilterNames::get().RATE_LIMIT);
```
Comment (zuercher) [inline test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc:341]: There's an #include of well_known_names.h at the top of this file that needs to be remove.

**Review by zuercher — COMMENTED** [review]
I noted the source of the compilation error below. Also, I think we can outright delete well_known_names.h in this PR as well. (I think that step may have been missed in some of your other PRs as well.)

File: test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc (hunk 1 above)
Comment (zuercher) [reply test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc:341]: > I think that step may have been missed in some of your other PRs as well.

Never mind that part. Just saw #16836.

Comment (daixiang0) [reply test/extensions/filters/network/thrift_proxy/filters/ratelimit/ratelimit_test.cc:341]: Thanks for review, I will fix them.

**Review by zuercher — APPROVED** [review]

**Pull request merged**